bin/install

```diff
@@ -18,23 +19,109 @@ error_exit() {
   exit "${2:-1}"
 }
 
-platform() {
+get_platform() {
   local -r os=$(
     case "$(uname | tr '[:upper:]' '[:lower:]')" in
       darwin) echo "osx" ;;
       *) echo "linux" ;;
     esac
   )
 
-  echo "${os}-amd64"
+  echo "${os}"
+  return
 }
+get_arch(){
+  declare arch="$(uname -m)"
+  if [ "$arch" == 'x86_64' ]; then
+    echo '64'
+  elif [ "$arch" == 'i386' ]; then
+    echo '32'
+  elif [ "$arch" == 'i686' ]; then
+    echo '32'
+  else
+    error_exit 'There are no official releases for your architecture'
+  fi
+  return
+}
+
+get_assets_url() {
```

Comment:
The function `get_assets_url(install_version)` returns the Github API endpoint that itself returns all available assets for a given release. This is done by querying the releases URL, extracting an array of all `'assets_url'` and `'tag_name'` properties within the json returned, putting them in two arrays, and correlating the two arrays.

Hacky solution, but it seems to work, given the absence of jq in the runtime system at the time of execution.

`error_exit`s on a release not found.

Have not tested what happens when the input is an empty string.